src/Neo.VM/ScriptBuilder.cs

```diff
@@ -114,7 +115,9 @@ public ScriptBuilder EmitPush(BigInteger value)
         /// <returns>A reference to this instance after the emit operation has completed.</returns>
         public ScriptBuilder EmitPush(bool value)
         {
-            return Emit(value ? OpCode.PUSH1 : OpCode.PUSH0);
+            Emit(value ? OpCode.PUSH1 : OpCode.PUSH0);
+            Emit(OpCode.CONVERT);
+            return EmitRaw(new ReadOnlySpan<byte>(new[] { (byte)StackItemType.Boolean }));
```

Comment:
Maybe it's cheaper to use [NOT](https://github.com/neo-project/neo/blob/fcbf479cb5522d532dcdf46c51e5baba7d47fda3/src/Neo/SmartContract/ApplicationEngine.OpCodePrices.cs#L178)